core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -624,7 +624,8 @@ public CreatePathResult createPath(RpcContext rpcContext, LockedInodePath inodeP
       }
 
       createdInodes.add(dir);
-      extensibleInodePath.getInodes().add(dir);
+      // No need to add to InodePath because we already add to its locklist
```

Comment:
How is it already in the locklist?

Reply:
This is not obvious, but line 521 `InodeLockList lockList = traversalResult.getInodeLockList();` actually gets the same locklist that is in the `LockedInodePath` . So when we use the lockList to lock new Inodes, it adds it to the `LockedInodePath`

Reply:
i agree using `extensibleInodePath.getLockList()` makes it less confusing. Not sure about `TraversalResult` returning `LockedInodePath`, I think there is a version of `traverseToInode` that does not take `LockedInodePath` as a parameter, it would be odd to return a `LockedInodePath` in that method.

Reply:
ack.